dataplane/forwarding/fwdaction/actions/select_action_list.go

```diff
@@ -130,12 +140,24 @@ func (*selectActionListBuilder) Build(desc *fwdpb.ActionDesc, ctx *fwdcontext.Co
 		return nil, fmt.Errorf("actions: Unable to find select function %v", s.hash)
 	}
 
+	allZeros := true
 	for _, l := range sal.Select.GetActionLists() {
 		a, err := fwdaction.NewActions(l.GetActions(), ctx)
 		if err != nil {
 			return nil, fmt.Errorf("actions: Unable to create actions %v, err %v", l, err)
 		}
 		s.set = append(s.set, a)
+		if l.GetWeight() != 0 {
+			allZeros = false
+		}
+		s.weights = append(s.weights, int(l.GetWeight()+s.weightSum))
+		s.weightSum += l.GetWeight()
```

Comment:
Hmm so since the AFT model is used for gRIBI the input nexthop weights can be as big as uint64, so there is a chance for overflow... Not sure if we want to actually write logic to handle this or should the RIB manager reject any such routes.

I think I'm ok with just a TODO here since this may not occur in practice.

Reply:
added todo